### ISSUE TITLE
Add font size controls to Page buttons

### ISSUE THREAD

**Issue by BBrazner** [bitfocus/companion#2748]
### Is this a feature relevant to companion itself, and not a module?

- [X] I believe this to be a feature for companion, not a module

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the feature

Currently, there is no option to control font size when editing the text on Page buttons. This feature request is to add font size options (e.g., Auto, 7pt, 14pt, etc.) to Page buttons.

**Comment by dnmeid** [comment]
All page names are available as internal variables like $(internal:page_number_1_name). You can replace the default page name button with a regular button and style it to your liking. Does that solve your problem?

**Comment by Julusian** [comment]
Perhaps we should look into whether we should remove these special button types, and turn them into some builtin 'presets' instead.

I suspect that making the page up and down buttons draw the same won't be possible, but the page name button seems much more doable

**Comment by Julusian** [comment]
It needs some internal changes, to expand the availability of the `$(this:page)` variables, as well as adding a new `$(this:page_name)` variable, then it is entirely possible to match the existing behavior of the page number button:



![Image](https://github.com/bitfocus/companion/assets/1327476/97c2dbe3-d95c-4ae2-8906-e808fea06f72)



I suspect that going forward we should make the decision between the showing the page name, or the page number be a manual choice, which will remove the need to use a feedback to do that here.  
In fact this may be desirable soon, as the changes in the experimental builds do nudge users to set page names more

Based on how this was done, maybe the page up and down buttons can be achieved too. I had to use a png for the yellow 'page' text, the same would work for page up and down. The symbols to use on the buttons could be handled via a couple of new internal variables

**Comment by dnmeid** [comment]
Seems I have to hurry up with the layers for button graphics. Then we can replicate the page button style exactly with a regular button and make all the properties available for customisation.

**Comment by Julusian** [comment]
@dnmeid yeah that sounds good. I'm not in a rush to change this, it is possible to make this custom button manually after all.

I ended up pushing the changes so that internal actions, feedbacks and button text can use `$(this:page) ` and related variables, so it can be written once and copied to pages that want it without reconfiguring.

**Comment by Jibun-no-Kage** [comment]
When is 3.4 expected to be released?

**Comment by Julusian** [comment]
I think we've done what we want with this now.
It isn't possible to edit the size directly, but the buttons can be 'edited' by one-click converting them into normal buttons (matching drawing and behaviour), so it is possible to manually tweak this at the cost of taking ownership of the config of the buttons.

And with #4212, needing to manually choose a size might no longer be necessary anyway.

So I'm going to close this, as there are solutions which may be sufficient. If you think this is still needed, or we need other configurablility of these buttons, please open a new issue
